kite-data/kite-data-crunch/src/main/java/org/kitesdk/data/crunch/DatasetTarget.java

```diff
@@ -99,12 +105,12 @@ public boolean handleExisting(WriteMode writeMode, long lastModForSource,
           LOG.info("Writing to existing dataset/view: " + view);
           break;
         case CHECKPOINT:
-          boolean ready = true; // dataset.isReady(); // TODO: add when CDK-451 is ready
           long lastModForTarget = -1;
           if (view instanceof LastModifiedAccessor) {
             lastModForTarget = ((LastModifiedAccessor) view).getLastModified();
           }
-          if (ready && lastModForTarget > lastModForSource) {
+
+          if (ready && (lastModForTarget > lastModForSource)) {
```

Comment:
> If you looked at modification time alone you could pick up incomplete output

Isn't this handled by the fact that your driver program will run the pipeline correctly? The modification time is only used to see whether the source data has been updated more recently than derived data. There's no chance that the derived data, which is calculated by the pipeline, is in the progress of being written.

What I meant by the "or" statement was that marking something ready updates its modification time. That means it is either ready or modified recently by simply checking the modification time in this case. Does that make sense?